jetty-client/src/main/java/org/eclipse/jetty/client/AbstractConnectionPool.java

```diff
@@ -281,16 +316,25 @@ protected void proceed()
 
     protected Connection activate()
     {
-        Pool<Connection>.Entry entry = pool.acquire();
-        if (entry != null)
+        while (true)
         {
+            Pool<ConnectionHolder>.Entry entry = pool.acquire();
+            if (entry == null)
+                return null;
+            Connection connection = entry.getPooled().connection;
+
+            long maxDurationNanos = this.maxDurationNanos;
+            if (maxDurationNanos > 0L && entry.getPooled().isExpired(maxDurationNanos))
+            {
+                IO.close(connection);
```

Comment:
Please add a debug statement before this line to log it's been closed due to expiration.

Reply:
Ok.